amads/pitch/midi2hz.py

```diff
@@ -0,0 +1,38 @@
+"""
+Provides the `midi2hz` function.
+
+Original doc: https://citeseerx.ist.psu.edu/document?repid=rep1&type=pdf&doi=6e06906ca1ba0bf0ac8f2cb1a929f3be95eeadfa#page=50
+"""
+
+def midi2hz(midi):
+    """
+    Converts a midi note number to the corresponding frequency in hz. 
+    Validates input to make sure that the input is non-negative. 
+
+    Parameters:
+    midi (float or a list of floats): The midi note number or a list of midi note numbers. 
+
+    Returns:
+    hz (float or a list of floats): The corresponding hz measurements for the midi note numbers.
+
+    Raises:
+    ValueError: If any midi note number is negative. 
+    """
+    
+    def convertMidi2Hz(singleMidiNoteNumber):
```

Comment:
Defining a function inside a function incurs unnecessary performance overhead, I wouldn't do that